src/unicode/word.rs

```diff
@@ -142,7 +137,7 @@ impl<'a> Iterator for WordIndices<'a> {
 
     #[inline]
     fn next(&mut self) -> Option<(usize, usize, &'a str)> {
-        while let Some((start, end, word)) = self.0.next() {
+        for (start, end, word) in self.0.by_ref() {
```

Comment:
This appears it can have the same `Iterator::find` simplification performed on it as the above hunk of the diff.